installinator/src/mock_peers.rs

```diff
@@ -228,18 +230,22 @@ impl PeersImpl for MockPeers {
         self.selected_peers.len()
     }
 
-    fn fetch_from_peer_impl(
+    async fn fetch_from_peer_impl(
         &self,
         peer: SocketAddrV6,
         // We don't (yet) use the artifact ID in MockPeers
         _artifact_hash_id: ArtifactHashId,
-        sender: FetchSender,
-    ) -> Pin<Box<dyn futures::Future<Output = ()> + Send>> {
+    ) -> Result<(u64, FetchReceiver), HttpError> {
         let peer_data = self
             .get(peer)
             .unwrap_or_else(|| panic!("peer {peer} not found in selection"))
             .clone();
-        Box::pin(async move { peer_data.send_response(sender).await })
+        let artifact_size = peer_data.artifact.len() as u64;
+
+        let (sender, receiver) = mpsc::channel(8);
+        tokio::spawn(async move { peer_data.send_response(sender).await });
+        // TODO: add tests to ensure an invalid artifact size is correctly detected
```

Comment:
Just checking - did you want to hit this TODO before merging? (I'm fine either way)

Reply:
No, it's not essential for PVT1 so I want to come back to it after that deadline.